src/main/java/edu/kit/kastel/sdq/artemis4j/grading/Course.java

```diff
@@ -0,0 +1,81 @@
+package edu.kit.kastel.sdq.artemis4j.grading;
+
+import edu.kit.kastel.sdq.artemis4j.ArtemisNetworkException;
+import edu.kit.kastel.sdq.artemis4j.LazyNetworkValue;
+import edu.kit.kastel.sdq.artemis4j.client.CourseDTO;
+import edu.kit.kastel.sdq.artemis4j.client.ExamDTO;
+import edu.kit.kastel.sdq.artemis4j.client.ProgrammingExerciseDTO;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A course, containing exercises.
+ */
+public class Course extends ArtemisConnectionHolder {
+    private final CourseDTO dto;
+    private final LazyNetworkValue<List<Exercise>> exercises;
+    private final LazyNetworkValue<List<Exam>> exams;
+
+    public Course(CourseDTO dto, ArtemisConnection connection) {
+        super(connection);
+
+        this.dto = dto;
+        this.exercises = new LazyNetworkValue<>(() -> {
+            List<Exercise> result = new ArrayList<>();
+
+            result.addAll(ProgrammingExerciseDTO.fetchAll(connection.getClient(), dto.id())
+                .stream()
+                .map(exerciseDTO -> new ProgrammingExercise(exerciseDTO, this))
+                .toList());
+
+            // TODO: add text exercises
```

Comment:
Don't forget TODO :)

Reply:
That's more of a long-term TODO, so I'll create an issue and remove the TODO